core/src/main/scala-3.0.0-RC3/cats/derived/show.scala

```diff
@@ -0,0 +1,31 @@
+package cats.derived
+
+import cats.Show
+import shapeless3.deriving.{Continue, K0, Labelling}
+
+object show extends ShowDerivation
+
+trait ShowDerivation:
+
+  extension (F: Show.type)
+    inline def derived[A](using gen: K0.Generic[A]): Show[A] = gen.derive(productShow, coproductShow)
+
+  given productShow[A](using inst : => K0.ProductInstances[Show, A], l: Labelling[A]): Show[A] with
+    def show(a: A) = {
+      var idx = 0
+      val x = inst.foldLeft[StringBuilder](a)(new StringBuilder(s"${l.label}("))(
+        [t] => (acc: StringBuilder, sh: Show[t], t0: t) => {
+          //elemLabels is backed by an array so this should be fast
+          val res = Continue(acc.append(s"${l.elemLabels(idx)}=${sh.show(t0)}, "))
+          idx = idx + 1
+          res
+          }
+      )
+      x.delete(x.length - 2, x.length).append(")")
```

Comment:
I think this is not safe for empty products

Reply:
🤦 good point!

Reply:
Isn't this just
```suggestion
      if (idx > 0) {x.delete(x.length - 2, x.length)}
```

Reply:
I should just give up on programming today 😂 Thanks